### ISSUE TITLE
[Ask] MLFlow Tagging and Experiments

### ISSUE THREAD

**Issue by yeelauren** [data-mermaid/mermaid-segmentation#34]
### Problem Statement

We need an automated or semi-automated way to track experiments in MLFlow between the classifier and the segmentation models.

### Desired Outcome

An organized way to tag, filter and identify our experiments using MLFlow 

https://mlflow.org/docs/latest/ml/tracking/tracking-api/#smart-tagging-for-organization

### Current State

Metrics: train/loss, train/accuracy, train/mean_iou, validation/accuracy, validation/mean_iou, checkpoint/accuracy, checkpoint/mean_iou, test/accuracy, test/mean_iou, train/time_taken

Params: num_classes, run_name, log_epochs, log_checkpoint, scalars from config.model.* and config.training.*

Artifacts: config/config.json, model checkpoint .pt

Tags: model_type, framework

### Success Criteria

Additional logging and metrics 
See also : https://github.com/data-mermaid/mermaid-classifier/issues/21

### Impact

Project organization
Experimentation and reproducibility 

### Metrics

- Per class iou, per class f1 already exist in [mermaidseg/model/eval.py](mermaidseg/model/eval.py),
- validation loss
- learning rate

### Parameters

- whitelist sources
- data padding
- batch size
- training mode (standard, concept, concept-bottle-neck)
- lambda_weight
- num_concepts
-

### System Metrics

- https://mlflow.org/docs/latest/ml/tracking/system-metrics/
- Helpful for:
- #30 
- #14 
- #17 

### Dataset

- https://mlflow.org/docs/latest/ml/dataset/

### Constraints

Storage for model checkpoints and model metadata
Removal of bad experiments 

### Solution Architecture

SageMaker, MLFlow, PyTorch
